docs/developer_guide/style_guide.rst

```diff
@@ -0,0 +1,28 @@
+Style Guide and General Conventions
+-----------------------------------
+
+As a general rule we use `black <https://black.readthedocs.io/en/stable/>`_ formater which should take care of most of
+the formating issues. To ensure that your commits are already formatted the repo already contains the machinery
+for per-commit hooks. To enable black formating do the following:
+
+1. Install pre-commit: :code:`pip install pre-commit`
+2. Execute :code:`pre-commit install` to install git hooks in your .git/ directory.
+
+For what is not covered by black, this project follows some of the conventions from `scikit-learn <https://scikit-learn.org/stable/>`_.
+
+#. Use underscores to separate words in non class names: :code:`n_samples` rather than :code:`nsamples`.
+#. Avoid multiple statements on one line. Prefer a line return after a control flow statement (if/for).
+#. Use relative imports for references inside the source code :code:`src/neuroconv`.
+#. Unit tests are an exception to the previous rule; they should use absolute imports, exactly as client code would.
+#. A corollary is that, if :code:`neuroconv.foo` exports a class or function that is implemented in :code:`neuroconv.foo.bar.baz`, the test should import it from :code:`neuroconv.foo`.
+#. Please dont use :code:`import *` in any case. It is considered harmful by the official Python recommendations. It makes the code harder to read as the origin of symbols is no longer explicitly referenced, but most important, it prevents using a static analysis tool like pyflakes to automatically find bugs in scikit-learn.
+#. Use the numpy `docstring standard <https://numpydoc.readthedocs.io/en/latest/format.html#numpydoc-docstring-guide>`_ in all your docstring.
```

Comment:
I'd recommend multi-section numbering or anything similar to differentiate these rules across the sections: https://neuroconv--28.org.readthedocs.build/en/28/developer_guide/style_guide.html#other-conventions

As it resets to `1.` each time it can get confusing which rule is being referenced in discussion

Reply:
Yes, I have been thinking on how to fix this since the numbering caused a previous confusion in the comments above. For some reason what I tried did not work out. After the updates are available https://github.com/catalystneuro/neuroconv/pull/53 I will try again to re-implement this to have a unique reference to each of the points.

Reply:
@CodyCBakerPhD I could not find something for this. If you see this in another documentation let me know and I will try to adapt it.

Reply:
For automation it looks like all you're missing is the master headers: https://stackoverflow.com/questions/2362197/restructuredtext-numbered-headers

Otherwise, I mean, you could just manually type out different headings...

1a.
1b.
1c.

2a.
2b.
2c.

etc...